tool/plugin/lib/plugin.dart

```diff
@@ -530,24 +530,44 @@ class BuildCommand extends ProductCommand {
       }
 
       // TODO(devoncarew): Remove this when we no longer support AS 3.1.
-      var processedFile = new File(
-          'flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java');
-      var oldSource;
-      if (spec.version == '2017.3') {
-        oldSource = processedFile.readAsStringSync();
-        var newSource = oldSource.replaceAll(
+      var processedFile1, processedFile2, oldSource1, oldSource2, newSource;
+      if ((spec.version == '2017.3' &&
+              (spec.isReleaseMode || Platform.environment['CI'] == "true")) ||
+          Platform.environment['DART_BOT'] == 'true') {
+        processedFile1 = new File(
+            'flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java');
+        oldSource1 = processedFile1.readAsStringSync();
+        newSource = oldSource1;
+        newSource = newSource.replaceAll(
           'import com.android.tools.idea.npw.model.NewModuleModel',
           'import com.android.tools.idea.npw.module.NewModuleModel',
         );
-        processedFile.writeAsStringSync(newSource);
+        newSource = newSource.replaceAll(
+          'import java.awt.*',
+          'import javax.swing.*',
+        );
+        newSource = newSource.replaceAll('public Image', 'public Icon');
+        newSource =
+            newSource.replaceAll('IconUtil.toImage', ''); // Leaves parens
+        processedFile1.writeAsStringSync(newSource);
+        processedFile2 = new File(
+            'flutter-studio/src/io/flutter/project/ChoseProjectTypeStep.java');
```

Comment:
Aside: should this file get renamed to `ChooseProjectTypeStep`?

Reply:
Yes, but fixing bad file names is outside the scope of this PR :)

Reply:
😃